assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -371,25 +407,37 @@ class EditPage extends Component {
 	}
 }
 
-export default withSelect( ( select, { clientId, attributes } ) => {
-	const { getMedia } = select( 'core' );
-	const { getBlockOrder, getBlocksByClientId, getBlockRootClientId } = select( 'core/block-editor' );
-
-	const innerBlocks = getBlocksByClientId( getBlockOrder( clientId ) );
-	const isFirstPage = getBlockOrder().indexOf( clientId ) === 0;
-	const isCallToActionAllowed = ! isFirstPage && ! innerBlocks.some( ( { name } ) => name === 'amp/amp-story-cta' );
-	const { getAnimatedBlocks } = select( 'amp/story' );
-
-	const { mediaId } = attributes;
-
-	const animatedBlocks = getAnimatedBlocks();
-	const animatedBlocksPerPage = ( animatedBlocks[ clientId ] || [] ).filter( ( { id } ) => clientId === getBlockRootClientId( id ) );
-	const totalAnimationDuration = getTotalAnimationDuration( animatedBlocksPerPage );
-	const totalAnimationDurationInSeconds = Math.ceil( totalAnimationDuration / 1000 );
-
-	return {
-		media: mediaId ? getMedia( mediaId ) : null,
-		allowedBlocks: isCallToActionAllowed ? ALLOWED_CHILD_BLOCKS : ALLOWED_MOVABLE_BLOCKS,
-		totalAnimationDuration: totalAnimationDurationInSeconds,
-	};
-} )( EditPage );
+export default compose(
+	withSelect( ( select, { clientId, attributes } ) => {
+		const { getMedia } = select( 'core' );
+		const { getBlockOrder, getBlocksByClientId, getBlockRootClientId, getBlock } = select( 'core/block-editor' );
+
+		const innerBlocks = getBlocksByClientId( getBlockOrder( clientId ) );
+		const isFirstPage = getBlockOrder().indexOf( clientId ) === 0;
+		const isCallToActionAllowed = ! isFirstPage && ! innerBlocks.some( ( { name } ) => name === 'amp/amp-story-cta' );
```

Comment:
Since this second check here does the same as `pageHasCTABlock` and is more robust (does not only check the last item in the list), let's use that logic for `pageHasCTABlock`.

Reply:
For determining `isLast` we still need to check if the CTA block is in the last position. Since that's needed for `isLast` only, I'll add this as a separate check then.

Reply:
Or well, it actually shouldn't need a separate check.